pandas-stubs/core/indexing.pyi

```diff
@@ -1,22 +1,20 @@
-from typing import (
-    Generic,
-    TypeVar,
-    Union,
-)
+from typing import TypeVar
 
 import numpy as np
 from pandas.core.indexes.api import Index
 
 from pandas._libs.indexing import _NDFrameIndexerBase
 from pandas._typing import (
+    MaskType,
     Scalar,
-    StrLike,
 )
 
-_IndexSliceT = TypeVar("_IndexSliceT", bound=Union[StrLike, Scalar, slice])
+_IndexSliceTuple = slice | tuple[Index | MaskType | Scalar | list[Scalar] | slice, ...]
+
+_IndexSliceTupleT = TypeVar("_IndexSliceTupleT", bound=_IndexSliceTuple)
 
-class _IndexSlice(Generic[_IndexSliceT]):
-    def __getitem__(self, arg) -> tuple[_IndexSliceT, ...]: ...
+class _IndexSlice:
+    def __getitem__(self, arg: _IndexSliceTupleT) -> _IndexSliceTupleT: ...
```

Comment:
Isn't it always a tuple, even when slicing 1d?  In my experiments I always say some soft of tuple whether slicing 1 level 2 or 3.

Reply:
Or is slice a subclass of tuple somehow?

Reply:
`_IndexSliceTuple` is either a `slice` or a `tuple`.  So if you do `pd.IndexSlice[1:2]`, that returns a `slice` not a `tuple`

Reply:
I must admit I have never directly used `pd.IndexSlice.__getitem__` - it seems to simply be an identity function.

```py
>>> obj = object()
>>> pd.IndexSlice[obj] is obj
True
```

Reply:
If you are doing slicing on a `MultiIndex`), it really becomes necessary.  See the docs at https://pandas.pydata.org/docs/reference/api/pandas.IndexSlice.html?highlight=indexslice#pandas.IndexSlice for a good example.